### PR TITLE
core/net/ip: Resolve DNS crash fix

### PR DESCRIPTION
**Pull request by mtusnio**

### CODE REVIEW

**Comment by thetoster** [comment]
Bart approves

**Comment by abhijit-mahajani-imgtec** [comment]
fix the typo in commit message

**Comment by mtusnio** [comment]
Fixed

File: core/net/ip/resolv.c [hunk 1]
```diff
@@ -954,7 +954,10 @@ newdata(void)
 #if !ARCH_DOESNT_NEED_ALIGNED_STRUCTS
     {
       static struct dns_answer aligned;
-      memcpy(&aligned, ans, sizeof(aligned));
+      uint8_t t;
```
Comment (manohar-narkhede-imgtec) [inline core/net/ip/resolv.c]: This workaround can be avoided if the structure dns_answer is defined with attribute "packed" as per below on line no. 232:

```
struct __attribute__ ((packed)) dns_answer {
  /* DNS answer record starts with either a domain name or a pointer
   * to a name already present somewhere in the packet. */
  uint16_t type;
  uint16_t class;
  uint16_t ttl[2];
  uint16_t len;
#if NETSTACK_CONF_WITH_IPV6
  uint8_t ipaddr[16];
#else
  uint8_t ipaddr[4];
#endif
};
```

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins - dev

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins - dev

**Comment by manohar-narkhede-imgtec** [comment]
Approved.

**Comment by nikhil-zinjurde-imgtec** [comment]
test with jenkins - dev

**Pull request merged**